### ISSUE TITLE
Plan program workflow

### ISSUE THREAD

**Issue by martysweet** [martysweet/cfn-lint#1]
COS
- Resource nesting is shown in the error messages
- Multiple error messages can be shown per run
- YAML and JSON are supported
- Uses the AWS API Spec for a knowledge base
- First release only checks Resources
- Give a link to the documentation for errors in an error summary once run
- Debug should throw errors when found, and again at the end in the summary

```
cfn-lint --debug --region eu-west-1 mytemplate.yaml
```

1. Validate users input
2. For each resource defined, generate Outputs depending on type defined, throw error if invalid type
3. Check for circular dependencies
4. Resolve all !Ref and !Attr calls, then perform !Join 
5. Recursively descend through the template (if valid type)
5.1. Take the resource type, store the resource name on a stack
5.2. Check parameters for properties are valid, descending into them if type is not string
5.3. Throw errors on type mismatch or where property is somethingArn and an Arn is not found to be it's value
